internal/parse/parts.go

```diff
@@ -1,155 +1,81 @@
 package parse
 
-import "fmt"
-
-// queryPart is an interface to be implemented by all parts.
+// A queryPart represents a section of a parsed SQL statement, which forms
+// a complete query when processed together with its surrounding parts, in
+// their correct order.
 type queryPart interface {
-	// String-like output for debugging purposes
+	// String returns the part's representation for debugging purposes.
 	String() string
 
-	// Translate part to its Sql equivalent
-	ToSql() string
+	// ToSQL returns the SQL representation of the part.
+	ToSQL() string
 }
 
-// qualifiedExpression represents a qualified expression
-// qualified. Examples are:
-//	... column as &Person ... 	<--- "column" is unqualified
-//	... t.column as &Person.name	<--- "name" and "column" and qualified
-type qualifiedExpression struct {
-	qualifier string
-	name      string
+type FullName struct {
+	Prefix, Name string
 }
 
-// InputPart represents an expression used as input in the SDL query.
-// Examples:
-//	$Address
-//	$Address.postal_code
+// InputPart represents a named parameter that will be send to the database
+// while performing the query.
 type InputPart struct {
-	goType
+	FullName
 }
 
 // NewInputPart is a constructor that returns a pointer to an InputPart.
 func NewInputPart(typeName string, tagName string) (*InputPart, error) {
```

Comment:
These constructors are not as vital now we have a simplified structure, a new InputPart can be instantiated straight from a FullName, which in all likely-hood is what we will have in the natural flow of the parser

Reply:
Done.
Removed constructors. Simplified tests.